t/bad_subclass.t

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env perl
+
+use lib 'lib', 't/lib';
+
+use Test2::V0;
+
+# This test verifies the error handling when a TCM class extends a
+# non-TCM subclass.  All this happens at compile time, so the usual
+# exception testing is not applicable.  The modules get a directory of
+# their own because they (intentionally) break TCM::Load.
+
+use lib 't/badlib';
+
+my $error;
```

Comment:
All of this code is very repetitive. I think you could make a sub that does most of this. Also, did you try using `dies` from `Test2::V0`? I think you could do something like:

```perl
is( dies { eval '...' }, q{}, 'Subclassing works fine' );
like( dies { eval '...' }, qr/.../, '...' );
```

Reply:
A sub can indeed do the job for the two tests which check the error message.  However, I doubt about `dies`: by using `eval`, I prevent the code from dying, so `dies` will always return `undef`, and then restore `$@` to its value before the call. I chose string `eval` because I need the `use` to happen at runtime, otherwise the tests can't catch it.  Test2::V0 has a `try_ok`, but no `try_fail`.  As an alternative, I could simulate `use` with `require` + `import`, which would then allow to use `dies`.  But then, I'd always ask myself why I'm testing a routine `import` which the users aren't supposed to call...

I'm currently implementing all the other suggestions, update to come soon...ish (midnight is approaching).